src/Doctrine/Odm/PropertyInfo/DoctrineExtractor.php

```diff
@@ -52,13 +52,71 @@ public function getProperties($class, array $context = []): ?array
         return $metadata->getFieldNames();
     }
 
+    public function getType($class, $property, array $context = []): ?Type
```

Comment:
I kinda want to mark this as internal as well ^^ also can't we type `$class` and `$property`?

Reply:
Should we mark the whole class internal here instead?